internal/handlers/handlers.go

```diff
@@ -1 +1,89 @@
 package handlers
+
+import (
+	"fmt"
+	"html/template"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Yandex-Practicum/go1fl-sprint6-final/internal/service"
+)
+
+// IndexHandler обрабатывает GET-запросы к корневой странице
+func IndexHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		http.Error(w, "Метод не поддерживается", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Читаем HTML-шаблон
+	t, err := template.ParseFiles("index.html")
+	if err != nil {
+		http.Error(w, "Ошибка загрузки шаблона", http.StatusInternalServerError)
+		return
+	}
+
+	err = t.Execute(w, nil)
+	if err != nil {
+		http.Error(w, "Ошибка рендеринга шаблона", http.StatusInternalServerError)
+	}
+}
+
+// UploadHandler обрабатывает загрузку файлов
+func UploadHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "POST" {
+		http.Error(w, "Метод не поддерживается", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Получаем загруженный файл
+	file, header, err := r.FormFile("myFile")
+	if err != nil {
+		http.Error(w, "Ошибка загрузки файла", http.StatusBadRequest)
+		return
+	}
+	defer file.Close()
+
+	// Читаем содержимое файла
+	content, err := io.ReadAll(file)
+	if err != nil {
+		http.Error(w, "Ошибка чтения файла", http.StatusInternalServerError)
+		return
+	}
+
+	// Конвертируем данные
+	convertedContent, err := service.DetectAndConvert(string(content))
+	if err != nil {
+		http.Error(w, "Ошибка конвертации", http.StatusInternalServerError)
+		return
+	}
+
+	// Создаем имя для выходного файла
+	timestamp := time.Now().UTC().String()
+	ext := filepath.Ext(header.Filename)
+	outputFilename := fmt.Sprintf("converted_%s%s", timestamp, ext)
+
+	// Создаем директорию для конвертированных файлов
+	convertedDir := "converted"
+	if err := os.MkdirAll(convertedDir, 0755); err != nil {
+		http.Error(w, "Ошибка создания директории", http.StatusInternalServerError)
+		return
+	}
+
+	// Формируем полный путь к файлу
+	outputPath := filepath.Join(convertedDir, outputFilename)
+
+	// Записываем результат
+	err = os.WriteFile(outputPath, []byte(convertedContent), 0644)
+	if err != nil {
+		http.Error(w, "Ошибка записи файла", http.StatusInternalServerError)
+		return
+	}
+
+	// Возвращаем результат
+	w.Header().Set("Content-Type", "text/plain")
```

Comment:
Чтобы не было кракозябр в браузере нужно так писать
```suggestion
	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
```